includes/sanitizers/class-amp-dev-mode-sanitizer.php

```diff
@@ -30,13 +32,17 @@ final class AMP_Dev_Mode_Sanitizer extends AMP_Base_Sanitizer {
 	 * @since 1.3
 	 */
 	public function sanitize() {
-		$this->dom->documentElement->setAttribute( AMP_Rule_Spec::DEV_MODE_ATTRIBUTE, '' );
+		$document_element = $this->dom->documentElement;
+
+		if ( ! DevMode::hasExemptionForNode( $document_element ) ) {
+			$document_element->setAttribute( AMP_Rule_Spec::DEV_MODE_ATTRIBUTE, '_amp_exempt' );
```

Comment:
Any objections to setting `data-ampdevmode` to `_amp_exempt` to mark it as set by the plugin?

Reply:
Humm. Would it be better to rather introduce a new property directly on the `Document` object to indicate whether the `documentElement` has always had the `data-ampdevmode` attribute.

In `\AmpProject\Dom\Document::loadHTMLFragment()` perhaps, it could do something like:

```php
$this->intialAmpDevMode = $this->hasAttribute( DevMode::DEV_MODE_ATTRIBUTE );
```

Then instead of checking for some special value, this flag could be checked. This would prevent a sanitizer from setting the value of the attribute to be this special`_amp_exempt` value.

So then the `\AmpProject\DevMode::hasExemptionForNode` method could be changed to be something like this:

```diff
diff --git a/lib/common/src/DevMode.php b/lib/common/src/DevMode.php
index 11ceeeb28..207e89a15 100644
--- a/lib/common/src/DevMode.php
+++ b/lib/common/src/DevMode.php
@@ -51,6 +51,12 @@ final class DevMode
             return false;
         }
 
+        // If the node is the documentElement and it did not originally have dev-mode, then count it as not being exempt.
+        $document = $node->ownerDocument;
+        if ( $document instanceof Document && $node === $document->documentElement && ! $document->intialAmpDevMode ) {
+            return false;
+        }
+
         return $node->hasAttribute(self::DEV_MODE_ATTRIBUTE);
     }
```

And then `DevMode::isExemptFromValidation()` could be continued to be used as before, rather than resurrecting the `is_exempt_from_validation` method.

Reply:
Done in https://github.com/ampproject/amp-wp/pull/4440/commits/f9296cab3c7f30426639fe14a836c2921607cba4.